onnxruntime/core/providers/nnapi/nnapi_builtin/builders/impl/reduction_op_builder.cc

```diff
@@ -0,0 +1,234 @@
+    } else {
+      // For case that `axes` is empty, act as an Identity op
+      const OperandType output_operand_type(operand_types.at(inputs[0].node_arg.Name()).type, input_shape);
+      model_builder.RegisterOperand(output, operand_indices.at(inputs[0].node_arg.Name()), output_operand_type);
+    }
+  }
+
+  return Status::OK();
+}
+
+// Operator support related
+
+int32_t ReductionOpBuilder::GetMinSupportedNNAPIFeatureLevel(
+    const NodeUnit& node_unit, const OpSupportCheckParams& /* params */) const {
+  const auto& op(node_unit.OpType());
+  if (op == "ReduceMean") {
+    return ANEURALNETWORKS_FEATURE_LEVEL_2;
+  }
+
+  return ANEURALNETWORKS_FEATURE_LEVEL_3;
+}
+
+bool ReductionOpBuilder::IsOpSupportedImpl(const InitializedTensorSet& initializers, const NodeUnit& node_unit,
+                                           const OpSupportCheckParams& /* params */) const {
+  const auto& inputs = node_unit.Inputs();
+  const auto& op(node_unit.OpType());
+
+  Shape input_shape;
+  if (!GetShape(inputs[0].node_arg, input_shape))
+    return false;
+
+  if (input_shape.size() > 4 || input_shape.empty()) {
+    LOGS_DEFAULT(VERBOSE) << "NNAPI reduction ops only support 1-4d shape, input is "
+                          << input_shape.size() << "d shape";
+    return false;
+  }
+
+  NodeAttrHelper helper(node_unit);
+
+  if (op == "ReduceMean") {
+    // ONNX `ReduceMean` will reduce by default all dimensions if axes is not provided. However, NNAPI does support the behavior
+    // to reduce all dimensions by default when 'axes' is empty/not provided.
+    // Notes from NNAPI doc:
+    // https://developer.android.com/ndk/reference/group/neural-networks#group___neural_networks_1ggaabbe492c60331b13038e39d4207940e0a047fe95a35b27f45c05432b6ca18eb6c
+    if (node_unit.SinceVersion() < 18 && !helper.HasAttr("axes")) {
+      // ONNX doc does not specify `axes` attribute to be optional, but we have unit test cases for ReduceMean 13- where axes attributes not provided
+      // and treated as default behavior to reduce all.
+      LOGS_DEFAULT(VERBOSE) << "For ReduceMean op version earlier than 18, `axes` is required to be provided as an attribute for NNAPI.";
+      return false;
+    }
+
+    if (node_unit.SinceVersion() > 13) {
+      if (inputs.size() > 1 && inputs[1].node_arg.Exists()) {
+        const auto& axes_name = inputs[1].node_arg.Name();
+        if (!Contains(initializers, axes_name)) {
+          LOGS_DEFAULT(VERBOSE) << "Axes of ReduceMean must be a constant initializer.";
+          return false;
+        }
+
+        // When `axes` is empty and `noop_with_empty_axes` equals 1, the case is supported and will be treated as an identity op.
+        if (initializers.at(axes_name)->int64_data_size() == 0 && helper.Get("noop_with_empty_axes", 0) == 0) {
+          LOGS_DEFAULT(VERBOSE) << "NNAPI ReduceMean doesn't support the behavior by default to reduce all dimensions when 'axes' is empty.";
```

Comment:
I think we can support this by creating an NNAPI input with all axes

Reply:
ok, got it. I think that makes sense. will update.